plugin/hook/server_test.go

```diff
@@ -0,0 +1,181 @@
+		ctx := context.Background()
+		mockPlugin := new(mock.HookPlugin)
+		hookSchemaErr := errors.New("error in hook schema")
+		// Mock.On("DoSomething").Return(errors.New("failed"))
+		mockPlugin.On("GetHookSchema", ctx, mock2.AnythingOfType("models.GetHookSchemaRequest")).Return(models.GetHookSchemaResponse{}, hookSchemaErr)
+
+		server := &GRPCServer{
+			Impl: mockPlugin,
+		}
+		request := &pb.GetHookSchema_Request{}
+		_, err := server.GetHookSchema(ctx, request)
+		assert.NotNil(t, err)
+		assert.Equal(t, hookSchemaErr, err)
+	})
+	t.Run("should return hook schema for request", func(t *testing.T) {
+		ctx := context.Background()
+		mockPlugin := new(mock.HookPlugin)
+		hookSchemaResponse := models.GetHookSchemaResponse{
+			Name:        "name",
+			Description: "description",
+			Image:       "image",
+			Type:        models.HookTypePost,
+			DependsOn:   []string{"depends-on1"},
+			SecretPath:  "secret-path",
+		}
+		mockPlugin.On("GetHookSchema", ctx, mock2.AnythingOfType("models.GetHookSchemaRequest")).Return(hookSchemaResponse, nil)
+
+		server := &GRPCServer{
+			Impl: mockPlugin,
+		}
+		request := &pb.GetHookSchema_Request{}
+		response, err := server.GetHookSchema(ctx, request)
+		assert.Nil(t, err)
+		assert.Equal(t, hookSchemaResponse.Name, response.Name)
+		assert.Equal(t, hookSchemaResponse.Type.String(), response.Type)
+	})
+}
+func TestGRPCServer_GetHookQuestions(t *testing.T) {
+	t.Run("should return error if hook questions returns error", func(t *testing.T) {
+		ctx := context.Background()
+		mockPlugin := new(mock.HookPlugin)
+		hookQuestionsErr := errors.New("error in hook ques")
+		// Mock.On("DoSomething").Return(errors.New("failed"))
+		mockPlugin.On("GetHookQuestions", ctx, mock2.AnythingOfType("models.GetHookQuestionsRequest")).Return(models.GetHookQuestionsResponse{}, hookQuestionsErr)
+		// .Run(func(args mock2.Arguments) {
+		// 	req := args.Get(0).(models.GetHookQuestionsRequest)
+		// 	assert.False(t, req.PluginOptions.DryRun)
+		// })
+
+		server := &GRPCServer{
+			Impl: mockPlugin,
+		}
+		request := &pb.GetHookQuestions_Request{
+			Options: &pb.PluginOptions{DryRun: false},
+		}
+		_, err := server.GetHookQuestions(ctx, request)
+		assert.NotNil(t, err)
+		assert.Equal(t, hookQuestionsErr, err)
+	})
+	//healthy
+	t.Run("", func(t *testing.T) {
+		ctx := context.Background()
+		mockPlugin := new(mock.HookPlugin)
+		hookQuestionResponse := models.GetHookQuestionsResponse{
+			Questions: models.PluginQuestions{},
+		}
+		mockPlugin.On("GetHookQuestions", ctx, mock2.AnythingOfType("models.GetHookQuestionsRequest")).Return(hookQuestionResponse, nil)
+
+		server := &GRPCServer{
+			Impl: mockPlugin,
+		}
+		request := &pb.GetHookQuestions_Request{}
+		response, err := server.GetHookQuestions(ctx, request)
+		assert.Nil(t, err)
+		assert.Equal(t, hookQuestionResponse.Questions, response.Questions)
+
+	})
+
+}
+
+func TestGRPCServer_ValidateHookQuestion(t *testing.T) {
```

Comment:
@mauliksoneji  can you help me with `func TestGRPCServer_ValidateHookQuestion` not able to identify what is the error (tried some changes in request (line 114) but not able to figure out). 
Also not able to figure out the how to check args (from line 103 )